### PR TITLE
Upgrade aws-sdk to gamma version

### PR DESCRIPTION
**Pull request by Amplifiyer**
_Issue #, if available:_ 
* #5472
* #5708 
* #5387 
* #5582 
* #5405
* #5320 
* #5332
* #5534
* #5393 

_Description of changes:_

**Note: The build won't succeed until aws-sdk release their gamma version. I've been testing with a local copy of cached verdaccio release.**

This PR is to upgrade aws-sdk version which fixes the above mentioned amplify issues. There are some compatibility changes since this release of aws-sdk is not 100% backwards compatible.

1. Browser environment now requires the http response to be either a blob or a stream. Updated axios http handler to now always request a blob and return it. This also makes it consistent for react-native and browsers environment to always return blobs to amplify customers. Need to discuss since this would be a breaking change. See table below for reference

Release | Browser | React Native
------------ | ------------ | -------------
Amplify V2 | UInt8Array | UInt8Array
Amplify V3 | String (#5393) | Blob
This PR | Blob | Blob

2. Local mock storage now requires another parameter `tls` to enforce http endpoints.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/storage/src/providers/detectReactNative.native.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export const isReactNative: boolean = true;
```
Comment (jordanranz) [inline packages/storage/src/providers/detectReactNative.native.ts:1]: Is this something we can benefit from in other categories?

Comment (Amplifiyer) [reply packages/storage/src/providers/detectReactNative.native.ts:1]: Realized that `Platform.isReactNative` already exists. Replacing this.

File: packages/storage/src/providers/AWSS3Provider.ts [hunk 2]
```diff
@@ -508,8 +509,9 @@ export class AWSS3Provider implements StorageProvider {
 		if (dangerouslyConnectToHttpEndpointForTesting) {
 			localTestingConfig = {
 				endpoint: localTestingStorageEndpoint,
-				s3BucketEndpoint: true,
-				s3ForcePathStyle: true,
+				tls: false,
+				bucketEndpoint: false,
```
Comment (Ashish-Nanda) [inline packages/storage/src/providers/AWSS3Provider.ts:513]: Why was bucketEndpoint changed to false?

**Comment by Amplifiyer** [comment]
> is to change the return type of `Storage.get` when `download: true` 

Thanks @elorzafe. The type that's changing is the result.Body. Is there any trick shortcut to specify a type definition that explicitly has one member of a specific type but otherwise generic?

File: packages/storage/src/providers/AWSS3Provider.ts (hunk 2 above)
Comment (Amplifiyer) [reply packages/storage/src/providers/AWSS3Provider.ts:513]: it's a change in the aws-sdk interpretation of `bucketEndpoint`

V3: https://github.com/aws/aws-sdk-js-v3/blob/c96f35f972c44706a391bb07e0a897e73b8d6dfe/packages/middleware-bucket-endpoint/src/bucketEndpointMiddleware.ts#L27

V2: https://github.com/aws/aws-sdk-js/blob/dfb49100f7db707a2479726f911ca31443f5a861/lib/services/s3.js#L404-L407

Comment (Ashish-Nanda) [reply packages/storage/src/providers/AWSS3Provider.ts:513]: Ok, makes sense!

Comment (sammartinez) [reply packages/storage/src/providers/AWSS3Provider.ts:513]: should we call this out in the readme?

Comment (Amplifiyer) [reply packages/storage/src/providers/AWSS3Provider.ts:513]: This is a 100% non-breaking change for Amplify customers

**Comment by masbaehr** [comment]
Just for reference: This update to gamma cause this critical issue, as all S3 uploads stopped working in Chrome:

https://github.com/aws-amplify/amplify-js/issues/5873

**Pull request merged**